cmd/prometheus/main.go

```diff
@@ -227,13 +227,15 @@ func reloadConfig(filename string, rls ...Reloadable) (err error) {
 		return fmt.Errorf("couldn't load configuration (-config.file=%s): %v", filename, err)
 	}
 
-	// Apply all configs and return the first error if there were any.
+	failed := false
 	for _, rl := range rls {
-		if err != nil {
-			err = rl.ApplyConfig(conf)
-		} else {
-			rl.ApplyConfig(conf)
+		if err := rl.ApplyConfig(conf); err != nil {
+			log.Error("Failed to apply configuration: ", err)
```

Comment:
If differentiating means that an error on initial load should exit the
application, then yes, we want that.

We do the same for bad configurations IIRC, so this would be consistent.

On Sun, Aug 14, 2016, 12:07 AM Julius Volz <notifications@github.com> wrote:

> In cmd/prometheus/main.go
> <https://github.com/prometheus/prometheus/pull/1890#discussion_r74692793>:
>
> >  	for _, rl := range rls {
> > -		if err != nil {
> > -			err = rl.ApplyConfig(conf)
> > -		} else {
> > -			rl.ApplyConfig(conf)
> > +		if err := rl.ApplyConfig(conf); err != nil {
> > +			log.Error("Failed to apply configuration: ", err)
>
> Hmm, wouldn't that also confuse people, because the initial config load is
> also just a reload internally? A startup config failure is probably the
> most common for users with a bad config. The question is whether it's worth
> differentiating the two cases in the log message. The caller already
> differentiates in the final log message (
> https://github.com/prometheus/prometheus/blob/master/cmd/prometheus/main.go#L120
> vs.
> https://github.com/prometheus/prometheus/blob/master/cmd/prometheus/main.go#L134-L144),
> so maybe that's enough?
>
> —
> You are receiving this because you were assigned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/prometheus/prometheus/pull/1890/files/4a866c13be444795ad30d8f8679bd9773be9cec7#r74692793>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEuA8nrnD-1qs_5jm4n5m5QTPSwCD6hFks5qfj_-gaJpZM4Jis_I>
> .
>